File: app/src/main/java/com/lamnt/foodorder/view/adapter/FoodsAdapter2.java
package com.lamnt.foodorder.view.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.lamnt.foodorder.R;

public class FoodsAdapter2 extends RecyclerView.Adapter<FoodsAdapter2.ViewHolder> {
    private LayoutInflater inflater;
    private Context context;

    public FoodsAdapter2(Context context) {
        this.context = context;
        inflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(inflater.inflate(R.layout.item_recent_restaurant, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Glide.with(context).load(R.drawable.demo_royal_tea).placeholder(R.drawable.ic_food).into(holder.imgFood);
    }

    @Override
    public int getItemCount() {
        return 6;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ImageView imgFood;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imgFood = itemView.findViewById(R.id.img_restaurant);
        }
    }
}

File: app/src/main/java/com/lamnt/foodorder/view/adapter/recycleradapter/FoodsAdapter2.kt
package com.lamnt.foodorder.view.adapter.recycleradapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.lamnt.foodorder.R

class FoodsAdapter2(private val context: Context) : RecyclerView.Adapter<FoodsAdapter2.ViewHolder>() {
    private val inflater: LayoutInflater = LayoutInflater.from(context)
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(inflater.inflate(R.layout.item_recent_restaurant, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        Glide.with(context).load(R.drawable.demo_royal_tea).placeholder(R.drawable.ic_food).into(holder.imgFood)
    }

    override fun getItemCount(): Int {
        return 6
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var imgFood: ImageView

        init {
            imgFood = itemView.findViewById(R.id.img_restaurant)
        }
    }

}